community/detectors/apache_http_server_cve_2021_41773/src/main/java/com/google/tsunami/plugins/detectors/rce/cve_2021_25646/ApacheHttpServerVulnDetector.java

```diff
@@ -0,0 +1,115 @@
+import com.google.tsunami.common.data.NetworkServiceUtils;
+import com.google.tsunami.common.net.http.HttpClient;
+import com.google.tsunami.common.net.http.HttpResponse;
+import com.google.tsunami.common.time.UtcClock;
+import com.google.tsunami.plugin.PluginType;
+import com.google.tsunami.plugin.VulnDetector;
+import com.google.tsunami.plugin.annotations.PluginInfo;
+import com.google.tsunami.proto.DetectionReport;
+import com.google.tsunami.proto.DetectionReportList;
+import com.google.tsunami.proto.DetectionStatus;
+import com.google.tsunami.proto.NetworkService;
+import com.google.tsunami.proto.Severity;
+import com.google.tsunami.proto.TargetInfo;
+import com.google.tsunami.proto.Vulnerability;
+import com.google.tsunami.proto.VulnerabilityId;
+import java.io.IOException;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Optional;
+
+/**
+ * A {@link VulnDetector} that detects the CVE-2021-41773 vulnerability.
+ */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "ApacheHttpServerVulnDetector",
+    version = "1.0",
+    description = "This detector checks for Apache HTTP Server 2.4.49 Path traversal and disclosure vulnerability.",
+    author = "threedr3am (qiaoer1320@gmail.com)",
+    bootstrapModule = ApacheHttpServerVulnDetectorBootstrapModule.class
+)
+public class ApacheHttpServerVulnDetector implements VulnDetector {
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  private final Clock utcClock;
+  private final HttpClient httpClient;
+
+  @Inject
+  ApacheHttpServerVulnDetector(@UtcClock Clock utcClock, HttpClient httpClient) {
+    this.utcClock = checkNotNull(utcClock);
+    this.httpClient = checkNotNull(httpClient);
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(NetworkServiceUtils::isWebService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    String targetUri =
+        NetworkServiceUtils.buildWebApplicationRootUrl(networkService) + "cgi-bin/.%2e/";
+    try {
+      HttpResponse response = httpClient.send(get(targetUri).withEmptyHeaders().build(),
+          networkService);
+      try {
+        Optional<String> server = response.headers().get("Server");
+        Optional<String> body = response.bodyString();
+        if (server.isPresent() && server.get().contains("Apache/2.4.49") && body.isPresent()
+            && !body.get().contains("You don't have permission to access this resource.")) {
+          return true;
+        }
+      } catch (Throwable t) {
```

Comment:
Can you be more specific on what exceptions you are catching here?

Reply:
Optional is used here, no try-catch is needed, deleted.